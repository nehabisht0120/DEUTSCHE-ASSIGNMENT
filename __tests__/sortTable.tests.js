// This is test file for row utils

"use strict";

const tableUtils = require("../es6/tableUtils/sortTable");

const mockedData =
  "<tbody>" +
  "<tr>" +
  "<th>name</th>" +
  "<th>bestBid</th>" +
  "<th>bestAsk</th>" +
  "<th>openBid</th>" +
  "<th>openAsk</th>" +
  `<th>lastChangeAsk</th>` +
  `<th>lastChangeBid</th>` +
  `<th>SparkLines</th>` +
  "</tr>" +
  '<tr id="gbpeur">' +
  "<td>gbpeur</td>" +
  "<td>1.28</td>" +
  "<td>1.33</td>" +
  "<td>1.27</td>" +
  "<td>1.30</td>" +
  "<td>0.02</td>" +
  "<td>0.02</td>" +
  "</tr>" +
  '<tr id="eusqw">' +
  "<td>eusqw</td>" +
  "<td>1.28</td>" +
  "<td>1.33</td>" +
  "<td>1.27</td>" +
  "<td>1.30</td>" +
  "<td>0.02</td>" +
  "<td>-0.01</td>" +
  "</tr>" +
  '<tr id="blahblah">' +
  "<td>blahblah</td>" +
  "<td>1.28</td>" +
  "<td>1.33</td>" +
  "<td>1.27</td>" +
  "<td>1.30</td>" +
  "<td>0.02</td>" +
  "<td>-0.01</td>" +
  "</tr>" +
  "</tbody>"; 


document.body.innerHTML = '<table id="stomp-table">' + mockedData +   "</table>";

test("it should sort the table with minimum absolute value", () => {
  // Set up our document body

  tableUtils.sortTable("stomp-table", 6);
  expect(document.getElementById("stomp-table").innerHTML).toMatchSnapshot();
});



