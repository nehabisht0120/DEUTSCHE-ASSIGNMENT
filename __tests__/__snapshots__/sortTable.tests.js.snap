// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should sort the table with minimum absolute value 1`] = `"<tbody><tr><th>name</th><th>bestBid</th><th>bestAsk</th><th>openBid</th><th>openAsk</th><th>lastChangeAsk</th><th>lastChangeBid</th><th>SparkLines</th></tr><tr id=\\"eusqw\\"><td>eusqw</td><td>1.28</td><td>1.33</td><td>1.27</td><td>1.30</td><td>0.02</td><td>-0.01</td></tr><tr id=\\"blahblah\\"><td>blahblah</td><td>1.28</td><td>1.33</td><td>1.27</td><td>1.30</td><td>0.02</td><td>-0.01</td></tr><tr id=\\"gbpeur\\"><td>gbpeur</td><td>1.28</td><td>1.33</td><td>1.27</td><td>1.30</td><td>0.02</td><td>0.02</td></tr></tbody>"`;
